numpy/core/src/multiarray/einsum.c.src

```diff
@@ -2135,7 +2135,7 @@ get_combined_dims_view(PyArrayObject *op, int iop, char *labels)
          * The char type may be either signed or unsigned, we
          * need it to be signed here.
          */
-        int label = (signed char)labels[idim];
+        int label = labels[idim];
```

Comment:
Why this change?

Reply:
@mattip we did that because it made it work on s390x and power9.
Of course it broke x86_64!
I thought I had reverted to a clean copy of master and made your change.
Now I think I made a MisTeAk :)
I will look at what I did and push again.